### ISSUE TITLE
bug: button in modal is not announced by VoiceOver

### ISSUE THREAD

**Issue by M2K-H5N** [ionic-team/ionic-framework#26156]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

After upgrading to Ionic 6, the iOS VoiceOver screen reader is ignoring any ion-button elements inside an ion-modal.
We are using Cordova to manage our hybrid applications.

- This worked previously on Ionic version 5.
- Any ion-button elements, not inside an ion-modal, are being treated as expected by the VoiceOver screen reader.
- The functionality still works fine with TalkBack on Android.

### Expected Behavior

When a user opens an Ionic Modal in an application on an iOS device, with the VoiceOver screen reader turned on, they should be able to navigate, by swiping right or left, to each element inside the modal, and the screen reader should read out the relevant details. This should include any ion-button elements, both in the header and the body of the modal. 

### Steps to Reproduce

**1. Create a simple Ionic Cordova seed app**

_ionic start myApp tabs --type=angular --cordova_

**2. Add the following HTML markup to "src/app/tab2/tab2.page.html" (under the ion-header, within the ion-content)**

`<ion-button id="open-modal" expand="block">Open</ion-button>`
`<ion-modal trigger="open-modal" (willDismiss)="onWillDismiss($event)">`
&nbsp;&nbsp;`<ng-template>`
&nbsp;&nbsp;&nbsp;&nbsp;`<ion-header>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-toolbar>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-buttons slot="start">`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-button (click)="cancel()">Cancel</ion-button>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`</ion-buttons>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-title>Welcome</ion-title>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-buttons slot="end">`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-button (click)="confirm()" [strong]="true">Confirm</ion-button>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`</ion-buttons>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`</ion-toolbar>`
&nbsp;&nbsp;&nbsp;&nbsp;`</ion-header>`
&nbsp;&nbsp;&nbsp;&nbsp;`<ion-content class="ion-padding">`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<div>Test Field 1</div>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-button (click)="cancel()">Test 1</ion-button>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<div>Test Field 2</div>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<ion-button aria-label="aria test" aria-hidden="false" role="button" tabindex="0" (click)="cancel()">Test 2</ion-button>`
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`<div>Test Field 3</div>`
&nbsp;&nbsp;&nbsp;&nbsp;`</ion-content>`
&nbsp;&nbsp;`</ng-template>`
`</ion-modal>`

**3. Add the following code to "src/app/tab2/tab2.page.ts"**

`onWillDismiss(event) {`
&nbsp;&nbsp;`console.log(event);`
`}`
`cancel() {`
&nbsp;&nbsp;`this.modal.dismiss(null, 'cancel');`
`}`
`confirm() {`
&nbsp;&nbsp;`this.modal.dismiss(null, 'confirm');`
`}`

**4. Add the iOS platform**

_ionic cordova platform add ios_

**5. Build the application**

_ionic cordova build ios_

### Ionic Info

**Ionic:**
   Ionic CLI                     : 6.20.3
   Ionic Framework               : @ionic/angular 6.3.2
   @angular-devkit/build-angular : 14.2.6
   @angular-devkit/schematics    : 14.2.6
   @angular/cli                  : 14.2.6
   @ionic/angular-toolkit        : 6.1.0
**Cordova:**
   Cordova CLI       : 11.0.0
   Cordova Platforms : ios 6.2.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 5.0.0, (and 3 other plugins)
**Utility:**
   cordova-res : not installed globally
   native-run  : 1.7.1
**System:**
   ios-deploy : 1.11.4
   ios-sim    : 8.0.2
   NodeJS     : v16.17.0
   npm        : 8.15.0
   OS         : macOS Monterey
   Xcode      : Xcode 14.0 Build version 14A5294g
**Angular**
Angular CLI: 14.2.6
Node: 16.17.0
Package Manager: npm 8.15.0 
OS: darwin x64
rxjs: 6.6.7
typescript : 4.7.
**Device**
iOS 16.0.2 (20A380)
Model: iPhone X (Model A1865, A1901, A1902, A1903)

### Additional Information

https://user-images.githubusercontent.com/116351616/197229899-51cf01df-0874-41ec-9757-19a69e18310a.mp4

**Comment by liamdebeasi** [comment]
Thanks for the report. We moved `role` onto the host of `ion-modal` which ended up triggering the WebKit bug that you are encountering.

I have reported this bug to the WebKit team: https://github.com/ionic-team/ionic-framework/commit/037d579b2a3a660358f1e9c9b020c9510bb9c6b0

I will see if we can move `role` back onto the inner div for now.
